### PR TITLE
hypre: E4S testsuite-inspired smoke tests

### PR DESCRIPTION
**Pull request by sshudler**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hypre/package.py [hunk 1]
```diff
@@ -176,6 +178,29 @@ def install(self, spec, prefix):
...
+
+                reason = "test: ensuring HYPRE examples run"
+                self.run_test('./ex5big', [], [], installed=True,
+                              purpose=reason, skip_missing=True, work_dir='.')
+                self.run_test('./ex15big', [], [], installed=True,
+                              purpose=reason, skip_missing=True, work_dir='.')
+
+                make("distclean")
+        else:
+            raise RuntimeError("MPI is required to run HYPRE smoke test")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hypre/package.py]: I think it's better to remove this. We don't want the tests to fail for `hypre~mpi`.

**Review by adamjstewart — COMMENTED** [review]
@ulrikeyang @osborn9 @balay can you review this?

File: var/spack/repos/builtin/packages/hypre/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/hypre/package.py]: I agree.  It would be better to "skip" the test if it does not apply.

**Comment by tldahlgren** [comment]
> @ulrikeyang @osborn9 @balay can you review this?

FWIW. I asked Sergei to add this test.  I will run it.

**Comment by tldahlgren** [comment]
Closing and re-opening PR to restart checks.

**Comment by sshudler** [comment]
@adamjstewart @tldahlgren  Removed the exception for `hypre~mpi`

**Review by osborn9 — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
LGTM

**Comment by tldahlgren** [comment]
Thank you for your contribution!

**Pull request merged**